pilot/pkg/config/kube/gateway/route_collections.go

```diff
@@ -63,7 +67,11 @@ func HTTPRouteCollection(
 						if m != nil {
 							r.Matches = []gateway.HTTPRouteMatch{*m}
 						}
-						if !yield(convertHTTPRoute(ctx, r, obj, n, !mesh)) {
+						istioRoute, ipCfg, configErr := convertHTTPRoute(ctx, r, obj, n, !mesh)
+						if istioRoute != nil && ipCfg != nil && ipCfg.enableExtProc {
+							routeRuleToInferencePoolCfg[istioRoute.Name] = ipCfg
```

Comment:
Mutating external state inside of an iterator seems possibly problematic. Not sure.